martha/martha_v3.js

```diff
@@ -406,23 +486,67 @@ async function marthaV3Handler(req, res) {
         return;
     }
 
-    let bondSA;
-    if (bondUrl && overlapFields(requestedFields, BOND_FIELDS)) {
+    // Try to retrieve the file name from the initial DRS response
+    let fileName = getDrsFileName(drsResponse);
+
+    // If we do NOT have the fileName yet, we will try to get that field from the accessUrl
+    const accessIdFields = fileName ? MARTHA_V3_ACCESS_ID_FIELDS : MARTHA_V3_ACCESS_ID_FN_FIELDS;
```

Comment:
To discuss: It should be possible to get the `fileName` only from the metadata without requesting the signed URL.

Reply:
Decision: Return `null` and do not access the signed URL to sleuth out the file name